android/app/src/main/java/covidsafepaths/bt/exposurenotifications/storage/Migration.kt

```diff
@@ -1,18 +1,32 @@
 package covidsafepaths.bt.exposurenotifications.storage
 
+import android.util.Log
 import io.realm.DynamicRealm
 import io.realm.FieldAttribute
 import io.realm.RealmMigration
+import io.realm.RealmObject
+import io.realm.RealmObjectSchema
 
-internal class Migration: RealmMigration {
-  override fun migrate(realm: DynamicRealm, oldVersion: Long, newVersion: Long) {
-    val schema = realm.schema
+internal class Migration : RealmMigration {
+    override fun migrate(realm: DynamicRealm, oldVersion: Long, newVersion: Long) {
+        Log.d("Migration", "Running migration from version $oldVersion to $newVersion");
+        val schema = realm.schema
+        var version = oldVersion
 
-    if (oldVersion == 1L) {
-      schema.create("ExposureEntity")
-        .addField("id", Long::class.java, FieldAttribute.PRIMARY_KEY)
-        .addField("dateMillisSinceEpoch", Long::class.java)
-        .addField("receivedTimestampMs", Long::class.java)
+        if (version == 1L) {
+            schema.create("ExposureEntity")
+                .addField("id", Long::class.java, FieldAttribute.PRIMARY_KEY)
+                .addField("dateMillisSinceEpoch", Long::class.java)
+                .addField("receivedTimestampMs", Long::class.java)
+            version++
+        }
+
+        if (version == 2L) {
+            schema.get("KeyValues")
+                ?.renameField("lastDownloadedKeyZipFileName", "value")
+                ?.removePrimaryKey()
+                ?.transform { obj -> obj.setString("id", KeyValues.LAST_PROCESSED_FILE_NAME_KEY) }
+                ?.addPrimaryKey("id")
+        }
```

Comment:
Do we need to worry about migrations yet given this isn't being used by real users?

Reply:
We don't need to worry, but I didn't want to force the QA team to re-install the app since they will lose all the keys they have (and will have to wait 24 hours to test again).